### PR TITLE
Update composer and psr2

### PR DESCRIPTION
**Pull request by cake17**
- psr2
- composer.json
- add editorconfig

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,44 +1,40 @@
...
+      "name": "Christian Winther",
+      "role": "Author",
+      "homepage": "http://cakephp.nu/"
+    }
+  ],
+  "require-dev": {
+    "cakephp/cakephp": "~3.0",
+    "friendsofcake/cakephp-test-utilities": "dev-master"
+  },
+  "suggest": {
```
Comment (jadb) [inline composer.json]: What for if empty?

File: composer.json [hunk 2]
```diff
@@ -1,44 +1,40 @@
...
+    }
+  },
+  "support": {
+    "source": "https://github.com/FriendsOfCake/search",
+    "issues": "https://github.com/FriendsOfCake/search/issues",
+    "wiki": "http://cakephp.nu/cakephp-search/",
+    "irc": "irc://irc.freenode.org/friendsofcake"
+  },
+  "extra": {
+    "installer-name": "Search"
```
Comment (jadb) [inline composer.json]: This isn't needed anymore.

**Comment by jadb** [comment]
:+1:

**Comment by jippi** [comment]
When @jadb 's comments has been fixed, it's totally mergable :)

**Comment by cake17** [comment]
@jadb @jippi corrections made :-)

**Pull request merged**